### ISSUE TITLE
Remove commons-logging from classpath

### ISSUE THREAD

**Issue by burkedavison** [GoogleCloudPlatform/spring-cloud-gcp#2000]
Pubsub-stream-binder and logging integration tests produce this message:
```
Standard Commons Logging discovery in action with spring-jcl: please remove commons-logging.jar from classpath in order to avoid potential conflicts
```
([Example log](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/actions/runs/5407413025/jobs/9825396164?pr=1933))

This dependency is brought in via spring-cloud-gcp-logging's  `com.google.cloud:google-cloud-logging-logback` dependency,
and spring-cloud-gcp-pubsub's `com.google.cloud:google-cloud-pubsub` dependency.

**Comment by zhumin8** [comment]
Adding a note: we should remove commons-logging from classpath for all artifacts published via SCGCP

**Comment by zhumin8** [comment]
fixed by #2407, and remaining cleanup after https://github.com/googleapis/google-http-java-client/pull/1905 has been released captured in #2415. Thus closing this one.
